### PR TITLE
remove NameSource window path for creating sources

### PR DESCRIPTION
**Pull request by andycreeth**
This removes a long-standing piece of tech debt that caused us to have 2 separate codepaths for adding a source that led to lots of replicated logic.

### CODE REVIEW

File: app/components/windows/AddSource.vue.ts [hunk 1]
```diff
@@ -44,7 +44,7 @@ export default class AddSource extends Vue {
     return { name: source.name, value: source.sourceId };
   });
 
-  selectedSourceId = this.sources[0].sourceId;
+  selectedSourceId = this.sources[0] ? this.sources[0].sourceId : null;
```
Comment (gettinToasty) [inline app/components/windows/AddSource.vue.ts]: does this mean `sources` is either an array or a single source? i think we should standardize our data shapes here to always be arrays if that's the case

Comment (andycreeth) [reply app/components/windows/AddSource.vue.ts]: PS this is the *one* thing I miss from CoffeeScript.  In CoffeeScript you can use the existential operator to skip null checks:
```CoffeeScript
selectedSourceId = this.sources[0]?.sourceId;
```

Comment (gettinToasty) [reply app/components/windows/AddSource.vue.ts]: there's a proposal for safe navigation in ES; i think it's going to be added somewhat soon

Comment (andycreeth) [reply app/components/windows/AddSource.vue.ts]: oh nice

**Pull request merged**